flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/SavepointWriter.java

```diff
@@ -78,12 +110,36 @@ public static SavepointWriter fromExistingSavepoint(String path, StateBackend st
         return new SavepointWriter(savepointMetadata, stateBackend);
     }
 
+    /**
+     * Creates a new savepoint. The savepoint will be written using the state backend defined via
+     * the clusters configuration.
+     *
+     * @param maxParallelism The max parallelism of the savepoint.
+     * @return A {@link SavepointWriter}.
+     * @see #newSavepoint(StateBackend, int)
+     * @see #withConfiguration(ConfigOption, Object)
+     */
+    public static SavepointWriter newSavepoint(int maxParallelism) {
+        Preconditions.checkArgument(
+                maxParallelism > 0 && maxParallelism <= UPPER_BOUND_MAX_PARALLELISM,
+                "Maximum parallelism must be between 1 and "
+                        + UPPER_BOUND_MAX_PARALLELISM
+                        + ". Found: "
+                        + maxParallelism);
+
+        SavepointMetadataV2 metadata =
+                new SavepointMetadataV2(
+                        maxParallelism, Collections.emptyList(), Collections.emptyList());
```

Comment:
I see that inside `SavepointMetadataV2` constructor, one list is defensively copied while another is not (`masterStates`). This may potentially cause `UnsupportedOperationException` if there are any attempts to modify it.